extensions/cl_ext_cxx_for_opencl.asciidoc

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018-2020 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+:data-uri:
+:icons: font
+include::../config/attribs.txt[]
+:source-highlighter: coderay
+
+= cl_ext_cxx_for_opencl
+:R: pass:q,r[^(R)^]
+Khronos{R} OpenCL Working Group
+
+== Name Strings
+
+`cl_ext_cxx_for_opencl`
+
+== Contact
+
+Please see the *Issues* list in the Khronos *OpenCL-Docs* repository: +
+https://github.com/KhronosGroup/OpenCL-Docs
+
+== Contributors
+
+Kevin Petit, Arm Ltd. +
+Sven Van Haastregt, Arm Ltd. +
+Anastasia Stulova, Arm Ltd. +
+Marco Antognini, Arm Ltd. +
+Neil Hickey, Arm Ltd. +
+
+== Notice
+
+include::../copyrights.txt[]
+
+== Status
+
+Draft
+
+== Version
+
+Built On: {docdate} +
+Version: 0.1.0
+
+== Dependencies
+
+This extension is written against the OpenCL Specification
+Version 3.0.2.
+
+This extension requires OpenCL 3.0 or OpenCL 2.x and `cl_khr_extended_versioning`.
+
+== Overview
+
+This extension adds support for building programs written using the C++ for
+OpenCL kernel language. It also enables applications to query the version of
+the language supported by the device compiler.
+
+== New build option
+
+This extension adds support for a new `CLC++` value to be passed to the
+`-cl-std` build option accepted by *clBuildProgram* and *clCompileProgram*.
+
+This option is only valid for programs created using
+*clCreateProgramFromSource*.
+
+== New API Enums
+
+Accepted value for the _param_name_ parameter to *clGetDeviceInfo*:
+
+[source,c]
+----
+CL_DEVICE_CXX_FOR_OPENCL_VERSION_EXT       0x4220
+----
+
+
+== Modifications to the OpenCL API Specification
+
+(Modify Section 4.2, *Querying Devices*) ::
++
+--
+
+(Add the following to Table 4.3, _Device Queries_) ::
++
+--
+
+[cols="1,1,4",options="header"]
+|====
+| cl_device_info
+| Return Type
+| Description
+
+| `CL_DEVICE_CXX_FOR_OPENCL_VERSION_EXT`
```

Comment:
Yes, this was a deliberate choice. Since the language is still young/experimental and no versions have been formally ratified, we thought reporting a single version made more sense. We don't want to set expectations with respect to support for older versions.

Reply:
Okay.

Reply:
However, if we use `CL_DEVICE_CXX_FOR_OPENCL_ALL_VERSIONS_EXT` we don't lose anything and in case we start to version the language we won't need to propose yet another extension or a version of an extension which is obviously more complicated for vendors and developers. It will also make things better aligned with the general direction.

Reply:
Discussed with Anastasia offline: we decided to report a single version as per my message above but rename the query to `CL_DEVICE_CXX_FOR_OPENCL_NUMERIC_VERSION_EXT` for consistency with `cl_khr_extended_versioning`.

Reply:
Okay, makes sense to me.  I'll resolve this.